source/extensions/filters/network/kafka/protocol_code_generator/kafka_generator.py

```diff
@@ -0,0 +1,422 @@
+#!/usr/bin/python
+
+# usage:
```

Comment:
If possible can you add some high level comments of what this does and why?
1) Where are definitions pulled from
2) What templates we use
3) What the output is, etc.